Week-4/session-11/oopsEs6/ abstractionAndEncpasulation.js

```diff
@@ -0,0 +1,59 @@
+class BankAccount {
+  #accountNumber;
+  #balance;
+  #accountHolderName;
+
+  constructor(accountNumber, balance, accountHolderName) {
+    this.#accountNumber = accountNumber;
+    this.#balance = balance;
+    this.#accountHolderName = accountHolderName;
+  }
+
+  deposit(amount) {
+    this.#balance += amount;
+  }
+
+  withdraw(amount) {
+    if (amount > this.#balance) {
+      console.log('Insufficient funds');
+    } else {
+      this.#balance -= amount;
+    }
+  }
+
+  getBalance() {
+    return this.#balance;
+  }
+}
+
+class CheckingAccount extends BankAccount {
+  constructor(accountNumber, balance, accountHolderName) {
+    super(accountNumber, balance, accountHolderName);
+  }
+}
+
+class SavingsAccount extends BankAccount {
+  constructor(accountNumber, balance, accountHolderName) {
+    super(accountNumber, balance, accountHolderName);
+  }
+
+  withdraw(amount) {
```

Comment:
you are calling super.withdraw(amount) without first checking if the withdrawal will result in a negative balance. You should include this check before proceeding with the withdrawal.